Documentation/CRDs/Cluster/network-providers.md

```diff
@@ -189,3 +259,333 @@ NAD is attached to the container, allowing the daemon to communicate with the re
+route between each other, the host must also be connected via Macvlan.
+
+Because the host IP range is a subset of the whole range, a route must be added to include the whole
+range.
+
+Such a configuration should be equivalent to the following:
+
+```console
+ip link add public-shim link eth0 type macvlan mode bridge
+ip addr add 192.168.252.${STATIC_IP}/22 dev public-shim
+ip link set public-shim up
+ip route add 192.168.0.0/16 dev public-shim
+```
+
+The NetworkAttachmentDefinition for the public network would look like the following, using
+Whereabouts' `exclude` option to simplify the `range` request. The Whereabouts `routes[].dst` option
+ensures pods route to hosts via the Multus public network.
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: public-net
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "192.168.0.0/16",
+        "exclude": [
+           "192.168.252.0/22"
+        ],
+        "routes": [
+          {"dst": "192.168.252.0/22"}
+        ]
+      }
+    }'
+```
+
+#### Macvlan, DHCP
+
+The network plan for this cluster will be as follows:
+- The underlying network supporting the public network will be attached to hosts at `eth0`
+- Macvlan will be used to attach pods to `eth0`
+- Pods and nodes will share the IP range 192.168.0.0/16
+- DHCP will be used to ensure nodes and pods get unique IP addresses
+
+Node configuration must allow nodes to route to pods on the Multus public network.
+
+Because pods will be connecting via Macvlan, and because Macvlan does not allow hosts and pods to
+route between each other, the host must also be connected via Macvlan.
+
+Such a configuration should be equivalent to the following:
+
+```console
+ip link add public-shim link eth0 type macvlan mode bridge
+ip link set public-shim up
+dhclient public-shim # gets IP in range 192.168.0.0/16
+```
+
+The NetworkAttachmentDefinition for the public network would look like the following.
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: public-net
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "dhcp",
+      }
+    }'
+```
+
+## Modifying CSI networking
+
+### Disabling Holder Pods with Multus and CSI Host Networking
+
+This migration section applies in the following scenarios:
+- CephCluster `network.provider` is `"multus"`, AND
+- Operator config `CSI_DISABLE_HOLDER_PODS` is changed to `"true"`, AND
+- Operator config `CSI_ENABLE_HOST_NETWORK` is (or is modified to be) `"true"`
+
+If the scenario does not apply, skip ahead to the
+[Disabling Holder Pods](#disabling-holder-pods) section below.
+
+**Step 1**
```

Comment:
Added simple **Step #** headers in response to a note Subham gave about having a hard time following.